### PR TITLE
altdns: init at unstable-2021-09-09

### PR DESCRIPTION
**Pull request by octodi**
## Description of changes

Added altdns [a DNS recon tool](https://github.com/infosec-au/altdns)

https://github.com/infosec-au/altdns/

Related to https://github.com/NixOS/nixpkgs/issues/81418

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/al/altdns/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+
+python3.pkgs.buildPythonApplication rec {
```
Comment (arthsmn) [inline pkgs/by-name/al/altdns/package.nix:6]:
```suggestion
python3.pkgs.buildPythonApplication {
```
It isn't being used.

File: pkgs/by-name/al/altdns/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  src = fetchFromGitHub {
+    owner = "infosec-au";
+    repo = "altdns";
+    rev = "8c1de0fa8365153832bb58d74475caa15d2d077a";
+    hash = "sha256-ElY6AZ7IBnOh7sRWNSQNmq7AYGlnjvYRn8/U+29BwWA=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [
+    setuptools
+    wheel
```
Comment (fabaff) [inline pkgs/by-name/al/altdns/package.nix:20]: Usually not needed for plain Python modules/tools.

```suggestion
    wheel
```

File: pkgs/by-name/al/altdns/package.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    termcolor
+    tldextract
+  ];
+
+  prePatch = ''
+  substituteInPlace requirements.txt \
+    --replace "argparse" "configargparse"
+  substituteInPlace setup.py \
+    --replace "argparse" "configargparse"
+  '';
```
Comment (fabaff) [inline pkgs/by-name/al/altdns/package.nix:35]: `argparse` is a standard module. At first glance the tool should work by simply removing `argparse` from the `requirements.txt` without pulling in `configargparse`.

File: pkgs/by-name/al/altdns/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  postInstall = ''
+    cp $src/words.txt $out/
```
Comment (fabaff) [inline pkgs/by-name/al/altdns/package.nix:38]: Isn't the content of that file covered by, what we already have available in, the `wordlists`package?

File: pkgs/by-name/al/altdns/package.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  '';
+
+  #Project has no tests
```
Comment (fabaff) [inline pkgs/by-name/al/altdns/package.nix:41]:
```suggestion
  # Project has no tests
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 280967` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>altdns</li>
    <li>altdns.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
@ofborg eval

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**